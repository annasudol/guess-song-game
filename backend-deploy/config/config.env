NODE_ENV=development
PORT=5000

DB_NAME=game

DB_USER=anna

DB_PASSWORD=a7QnAkEqhr0hlrC6