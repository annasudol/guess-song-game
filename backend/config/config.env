NODE_ENV=development
PORT=4000

DB_NAME=game

DB_USER=anna

DB_PASSWORD=a7QnAkEqhr0hlrC6