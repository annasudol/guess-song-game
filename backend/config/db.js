/* eslint-disable @typescript-eslint/no-var-requires */
const mongoose = require('mongoose');

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
const connectDB = async () => {
  // eslint-disable-next-line no-undef
  const conn = await mongoose.connect(`mongodb+srv://${process.env.DB_USER}:${process.env.DB_PASSWORD}@cluster0-i8t1t.mongodb.net/${process.env.DB_NAME}?retryWrites=true&w=majority`, {
    useUnifiedTopology: true,
    useNewUrlParser: true,
    useCreateIndex: true,
  });

  console.log(`MangoDB connected: ${conn.connection.host}`);
};

// eslint-disable-next-line no-undef
module.exports = connectDB;
