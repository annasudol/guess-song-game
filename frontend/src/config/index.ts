export * from './routes';
export * from './hash';
