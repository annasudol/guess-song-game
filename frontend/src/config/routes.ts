const scopes = ['user-top-read', 'user-read-currently-playing', 'user-read-playback-state'];

const spotifyAuth = 'https://accounts.spotify.com/authorize';

export const AppRoutes = {
  Authorize: `${spotifyAuth}?client_id=${
    process.env.REACT_APP_spotifyID
    }&redirect_uri=http://localhost:${process.env.REACT_APP_redirectPORT}/playlist/&scope=${scopes.join('%20')}&response_type=token&show_dialog=true`,
  Game: '/game/:playlistID',
  Login: '/',
  Playlists: '/playlist',
  Summary: '/summary',
  Scores: '/scores',
};
