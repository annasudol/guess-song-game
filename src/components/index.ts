export * from './alert/Alert';
export * from './button/Button';
export * from './button/ButtonLogout';
export * from './image/Image';
export * from './game/Game';
export * from './login/Login';
export * from './listSongs/ListSongs';
export * from './navigation/Navigation';
export * from './errorMessage/ErrorMessage';
export * from './playlists/Playlists';
export * from './playListItem/PlaylistItem';
export * from './scoreInfo/ScoreInfo';
export * from './scores/Scores';
export * from './summary/Summary';
export * from './search/Search';
export * from './song/Song';
export * from './timer/Timer';
