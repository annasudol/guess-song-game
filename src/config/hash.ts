const hash:
  | {}
  | {
      access_token: string;
      token_type: string;
      expires_in: '3600';
    } = window.location.hash
  .substring(1)
  .split('&')
  .reduce(function(initial, item) {
    if (item) {
      const parts = item.split('=');

      initial[parts[0]] = decodeURIComponent(parts[1]);
    }

    return initial;
  }, {});

window.location.hash = '';
export default hash;
