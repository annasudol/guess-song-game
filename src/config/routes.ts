const scopes = ['user-top-read', 'user-read-currently-playing', 'user-read-playback-state'];

const spotifyAuth = 'https://accounts.spotify.com/authorize';

export const AppRoutes = {
  Authorize: `${spotifyAuth}?client_id=${
    process.env.REACT_APP_spotifyID
    }&redirect_uri=${process.env.REACT_APP_redirectUri}&scope=${scopes.join('%20')}&response_type=token&show_dialog=true`,
  Game: '/game/:playlistID',
  Login: '/',
  Playlists: '/playlist',
  Summary: '/summary',
  Scores: '/scores',
};
