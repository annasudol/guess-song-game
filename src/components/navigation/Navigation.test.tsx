import { shallow } from 'enzyme';
import React from 'react';
import { AppRoutes } from '../../config';
import toJson from 'enzyme-to-json';
import { Navigation } from '../../components';
jest.mock('react-redux');
import * as redux from 'react-redux';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: (): jest.Mock<any, any> => mockDispatch,
  useSelector: jest.fn(),
}));

describe('component', (): void => {
  describe('Navigation', (): void => {
    beforeEach(() => {
      const spy = jest.spyOn(redux, 'useSelector');
      spy.mockReturnValue({ name: 'Test' });
    });
    it('match snapshot', (): void => {
      const wrapper = shallow(<Navigation />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('renders NavLink with correct path', (): void => {
      const wrapper = shallow(<Navigation />);
      expect(wrapper.find('NavLink').prop('to')).toBe(AppRoutes.Playlists);
    });
    it('displays user name', (): void => {
      const wrapper = shallow(<Navigation />);
      expect(wrapper.find('span').text()).toBe('hello Test!');
    });
    it('calls clear function upon clicking navigation button', (): void => {
      const wrapper = shallow(<Navigation />);
      wrapper.find('NavLink').simulate('click');

      expect(mockDispatch).toHaveBeenCalledTimes(1);
    });
  });
});
