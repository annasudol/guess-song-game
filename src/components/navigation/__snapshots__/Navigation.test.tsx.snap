// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Navigation match snapshot 1`] = `
<div
  className="nav bg-pink-900"
>
  <div
    className="container flex justify-between relative"
  >
    <NavLink
      onClick={[Function]}
      to="/playlist"
    >
      <Image
        alt="logo"
        className="mt-2"
        size="small"
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "logo.svg",
          }
        }
      />
    </NavLink>
    <div
      className="flex justify-between h-full w-60 mr-4"
    >
      <span
        className="font-alba text-light-gray self-center text-sm md:text-lg pt-4 pr-2"
      >
        hello 
        Test
        !
      </span>
      <Link
        className="font-alba text-orange-500 self-center text-sm md:text-lg pt-4 pr-2"
        to="/scores"
      >
        your scores
      </Link>
      <ButtonLogout />
    </div>
  </div>
</div>
`;
